agent/api/task.go

```diff
@@ -590,6 +596,27 @@ func (task *Task) shouldOverrideNetworkMode(container *Container, dockerContaine
 	return true, networkModeContainerPrefix + pauseContainer.DockerID
 }
 
+// overrideDNS overrides a container's host config if the following conditions are
+// true:
+// 1. Task has an ENI associated with it
+// 2. ENI has custom DNS IPs and search list associated with it
+// This should only be done for the pause container as other containers inherit
+// /etc/resolv.conf of this container (they share the network namespace)
+func (task *Task) overrideDNS(hostConfig *docker.HostConfig) *docker.HostConfig {
+	eni := task.GetTaskENI()
+	if eni == nil {
+		return hostConfig
+	}
+	if len(eni.DomainNameServers) > 0 {
+		hostConfig.DNS = eni.DomainNameServers
```

Comment:
Not sure if we need add logs for these two check, if not, then do we still need the check?

Reply:
I don't think we need to log anything here. Why wouldn't we need these checks though?

Reply:
Won't this has the same effect, we can even assign the empty value when the `len(xxx)==0`, any side effect?
```
hostConfig.DNS = eni.DomainNameServers
hostConfig.DNSSearch = eni.DomainNameSearchList
```

Reply:
Oh I didn't consider that. Yeah, that's effectively doing the same thing. I'll update this.